lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1347,9 +1426,26 @@ class MongoClientInterface {
      * @param {Logger}log logger instance
      * @param {Function} cb callback containing error
      * and BulkWriteResult
+     * @param {object} [params] request params
      * @return {undefined}
      */
-    internalDeleteObject(collection, bucketName, key, filter, log, cb) {
+    internalDeleteObject(collection, bucketName, key, filter, log, cb, params = null) {
+        // filter used when deleting object
+        const deleteFilter = Object.assign({
+            _id: key,
+        }, filter);
+
+        if (params && params.shouldOnlyDelete) {
```

Comment:
i fear `shouldOnlyDelete` is not very clear about what happens: I would prefer expliciting the specific behavior we (sometimes) request, e.g. sometimes like `needObjectDeletedOplog`

Reply:
So you want to invert the default behavior and only do the extra calls to MongoDB when we request it? How to detect when we "need" it?

Reply:
I updated the flag with a more meaningful name without changing the logic: https://github.com/scality/Arsenal/pull/2134/commits/cc75b89115a7a8d4f26417fde5103f80421bedef#diff-9c207befbd8f4d79a3b6e3d7d143444e804ca1684236b8fd9fa0f9e03c53477aR1439

Let me know if you want a better name or something else...